src/core/dbus-execute.c

```diff
@@ -2389,19 +2389,28 @@ int bus_exec_context_set_transient_property(
                         return r;
 
                 while ((r = sd_bus_message_read(message, "(ssbt)", &source, &destination, &ignore_enoent, &mount_flags)) > 0) {
+                        char *s, *d;
```

Comment:
[`bind_mount_add` ](https://github.com/systemd/systemd/blob/a9586f0ca87b8a3ce880313fb309cb8ee923638f/src/core/namespace.c#L1584) is already doing a `strdup` on both `source` and `destination`.
Why were the initial `strdup`s needed in the first place?
Why don't you just remove the two `strdup`?

(Even if it was needed, you forgot to use `_cleanup_free_` here.
But even with `_cleanup_free_`, you might still leak memory if `bind_mount_add` fails due to the `TAKE_PTR` that puts `s` and `d` to `NULL` before knowing whether `bind_mount_add` succeeded.)

Reply:
Thanks for the review. `_cleanup_free_` indeed matches other code here.

Regarding setting pointers to NULL before the contents is freed, this seems to be happening [here](https://github.com/systemd/systemd/blob/12dcab2f9ece9b101a38b8c676d84be6207629ce/src/core/dbus-execute.c#L1493) too.

Also [this](https://github.com/systemd/systemd/blob/12dcab2f9ece9b101a38b8c676d84be6207629ce/src/core/dbus-execute.c#L1865) looks like the `s` would not be freed in case of error. There are a lot of similar cases.

Reply:
Sorry I must have been unclear, what I meant is that you should not even do `strdup`.
Based on what [`bind_mount_add`](https://github.com/systemd/systemd/blob/a9586f0ca87b8a3ce880313fb309cb8ee923638f/src/core/namespace.c#L1584) does, there is no need to `strdup` the `source` and `destination`. So  it removes the need to do a OOM check as well.